sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -401,7 +402,7 @@ class SQLContext private[sql](
    */
   @Experimental
   def createDataFrame[A <: Product : TypeTag](rdd: RDD[A]): DataFrame = {
-    SparkPlan.currentContext.set(self)
+    SparkPlan.currentContext.set(new WeakReference[SQLContext](self))
```

Comment:
A `WeakReference` is quite weak; you'll lose the reference on a full GC. Is this not going to cause the code to fail suddenly? I don't know that references are any remedy for a leak.